grammars/php.cson

```diff
@@ -3724,3 +3734,32 @@
         ]
       }
     ]
+  
+  'ternary_shorthand':
+    'match': '\\s*(\\?:)\\s*'
+    'captures':
+      '1':
+        'name': 'keyword.operator.ternary.php'
+
+  'ternary_expression':
+    'begin': '(?<!\\?)\\s*(\\?)\\s*(?!:|\\?)'
```

Comment:
I think this can be simplified to a simple `\\?` if you include `#ternary_shorthand` and `#null_coalescing` _before_ `ternary_expression`.

Reply:
Would that create a more fragile situation though? Thought about that as well but I went this route to keep it future proof.

Reply:
@Ingramz what do you think about this?

Reply:
@50Wliu To my knowledge, this is that last thing that needs addressing with this PR, correct?

Reply:
Yep, I'll bring it up with him.

Reply:
I'm sorry, it got buried by GitHub marking it "outdated".

I vote for reordering these rules. It will make the rules more fragile if they are viewed independently from eachother and if you wanted to include them selectively, but if they are grouped together (which is currently the case and most likely will stay this way), the two other rules do the same job of asserting the negative lookahead/lookbehind of the third rule without the need of repeating ourselves. It's as efficient or better in terms of matching and should be easier to read for anyone seeing this for the first time.

Reply:
Works for me! I'll adjust later on this afternoon 👍 

Re: GitHub marking outdated.... I can relate. That drives me nuts!

Reply:
@50Wliu All set now. To my knowledge I've now addressed everything that was brought up.